### ISSUE TITLE
Bug: custom element content gets rendered as part of the output

### ISSUE THREAD

**Issue by gluck** [preactjs/preact-custom-element#18]
Minimal reproduction sandbox: https://codesandbox.io/s/preact-custom-element-rendering-content-issue-pkr70

custom element content/children should only be rendered if the custom element hasn't been defined (or browser doesn't support CE).
Also related, things get really nasty if you try to add `{children}` to the component output (duplicated nested content).

**Comment by developit** [comment]
This is fixed by #23 and upgrading to Preact X. Thanks for reporting!

Working upgraded version of the sandbox showing that `{children}` is also now fixed: https://codesandbox.io/s/preact-custom-element-rendering-content-issue-vyj8s

With the new approach, `children` is only populated once based on the initial DOM tree. Attribute updates don't re-scan children, so preact-rendered children aren't re-included anymore.

**Comment by amindunited** [comment]
@developit I was playing around with the sandbox that you shared, and it seems that any change to the return value of "Greeting" will cause duplication of the children.
for example simply changing the h1 to an h2:
```
const Greeting = ({ children, name = "World" }) => [
  <h2 id="hello">Hello {name}!</h2>,
  <fieldset style={{ zoom: 0.5 }}>children = {children}</fieldset>
]
```
...will duplicate the textNode from the HTML. 😢

Also thanks for looking at this, and I really like the refactor! 😄
